vcs/src/main/java/org/openjdk/skara/vcs/DiffComparator.java

```diff
@@ -22,12 +22,16 @@
  */
 package org.openjdk.skara.vcs;
 
-import java.io.*;
-import java.nio.file.*;
 import java.util.*;
 import java.util.regex.Pattern;
 
 public class DiffComparator {
+
+    private static final Pattern COPYRIGHT_PATTERN = Pattern.compile("""
+            -(.)*Copyright \\(c\\) (?:\\d|\\s|,)* Oracle and/or its affiliates\\. All rights reserved\\.
+            \\+(.)*Copyright \\(c\\) (?:\\d|\\s|,)* Oracle and/or its affiliates\\. All rights reserved\\.
```

Comment:
If the only difference between these lines is the first character, we should only need one line.
```suggestion
            [-+](.)*Copyright \\(c\\) (?:\\d|\\s|,)* Oracle and/or its affiliates\\. All rights reserved\\.
```

Reply:
But if we only use one line like you suggested, this will introduce a problem. In the backport PR, the user could make any change to the first line of copyright.  For example, the user could change the first line to "Random line" and the backport pr would still be considered as clean backport.  

Because the hunk will be 
```
@@ -2,1 +2,1 @@
- * Copyright (c) 2020, 2023, Oracle and/or its affiliates. All rights reserved.
+ * Random line.
```
and this hunk would match the regex pattern and be filtered out.


I think we still need two lines to make sure the user **only** changes the copyright years.

Reply:
Ah I get it now, this is a multiline pattern, not an either this line or that line. I thought you needed to tell something in the Java regex API that it was multiline, but maybe you don't if the pattern has a newline in it?

Reply:
I think this should work as is. The `Pattern.MULTILINE` and `Pattern.DOTALL` are used to change the meaning of `^`, `$` and `.` respectively to work differently with newlines. I don't think that's relevant for this pattern which contains an explicit newline. Sorry for the confusion!

Reply:
I think having a newline in the pattern is enough, so we don't need to tell the java regex api that it is a multiline pattern.